src/xcode/ENA/ENA/Source/Scenes/QRScanner/Model/Parser/QRCodeParser.swift

```diff
@@ -60,15 +61,18 @@ class QRCodeParser: QRCodeParsable {
 			)
 		}
 
-		guard let parser = parser else {
+		guard parser != nil else {
```

Comment:
Why not `guard let`?

Leaving the scope of the function should dealloc the instance as well, without having the need to set it to `nil` explicitly